presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -104,6 +107,20 @@ public SystemAccessControl create(Map<String, String> config)
         }
     }
 
+    @VisibleForTesting
+    static FileBasedSystemAccessControlRules parse(byte[] json)
+    {
+        ObjectMapper mapper = new ObjectMapperProvider().get()
+                .enable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        Class<FileBasedSystemAccessControlRules> javaType = FileBasedSystemAccessControlRules.class;
```

Comment:
You can use `JsonCodec`:
```java
return new JsonCodecFactory().jsonCodec(FileBasedSystemAccessControlRules.class).fromJson(json);

Reply:
I wish I could. Notice that what you proposed is more or less the same as it used to be.

The crucial part here is to `.enable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)` to make json parser fail when it finds any unknown property.

Using `jsonCodec` for protocol is great. However, for parsing configuration we should not allow anything unrecognized be placed in the file, exactly like we do for property files.

I was also thinking about changing `JsonCodecFactory`, but so far I do not have anything brilliant. Maybe I could add `jsonConfigurationCodec` which would return `JsonCodec` with enabed `DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES`. What do you think?

How many json configuration files do we have? It would be nice to fix all such places.